File: 724_find_pivot_index.py
from typing import List

class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        leftCount = 0
        totalCount = sum(nums)
        for i in range(0,len(nums)):
            rightCount = totalCount - leftCount - nums[i]
            if(leftCount == rightCount):
                return i
            leftCount = leftCount + nums[i]
        return -1
       

File: 724_find_pivot_index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function (nums) {
  const totalCount = nums.reduce((a, b) => a + b);
  let leftCount = 0;
  for (let index = 0; index < nums.length; index++) {
    let rightCount = totalCount - leftCount - nums[index];
    if (leftCount === rightCount) {
      return index;
    }
    leftCount += nums[index];
  }
  return -1;
};